.github/workflows/release.yml

```diff
@@ -0,0 +1,182 @@
+name: Release
+on:
+  workflow_dispatch:
+    inputs:
+      increment:
+        description: "How to increment version (use 'version' to specify version)"
+        required: true
+        type: choice
+        default: minor
+        options:
+          - major
+          - minor
+          - patch
+          - version
+      version:
+        description: >
+          The full version to release (first choose 'version' from the
+          'increment' dropdown)
+        required: false
+        type: string
+permissions:
+  contents: write
+jobs:
+  release:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/create-github-app-token@v2
+        id: app-token
+        with:
+          app-id: ${{ vars.QLTY_APP_ID }}
+          private-key: ${{ secrets.QLTY_APP_PRIVATE_KEY }}
+          owner: ${{ github.repository_owner }}
+
+      - name: Checkout
+        uses: actions/checkout@v4
```

Comment:
These action references will need to be changed from @vX tags to commit SHAs but that can be done as follow up